docs: fix license specifier and add hint to spdx (#664)
prefix-dev/rattler-build: docs/recipe_file.md

  documentation: https://docs.com
```

> 1.  These follow the SPDX specifiers more info here: [SPDX](https://spdx.org/licenses/)

Reviewer: ```suggestion
1.  Only the SPDX specifiers are allowed, more info here: [SPDX](https://spdx.org/licenses/)
```
